### PR TITLE
Pay UI improvements 

### PR DESCRIPTION
**Pull request by MananTank**
## Problem solved

Short description of the bug fixed or feature added

## Changes made

- [ ] Public API changes: list the public API changes made if any
- [ ] Internal API changes: explain the internal logic changes

## How to test

- [ ] Automated tests: link to unit test file
- [ ] Manual tests: step by step instructions on how to test

## Contributor NFT

Paste in your wallet address below and we will airdrop you a special NFT when your pull request is merged. 

```Address: ```

---

## PR-Codex overview
This PR focuses on Pay UI improvements and breaking down complex screens into multiple simpler ones. It also includes enhancements like style adjustments, new components, and test updates.

### Detailed summary
- Added `getBuyTokenAmountFontSize` function for dynamic font size calculation
- Updated UI components like `ModalHeader` and `PayTokenIcon`
- Improved wallet connection setup in tests with `SetConnectedWallet`
- Adjusted styles in `Button` component based on the theme colors

> The following files were skipped due to too many changes: `packages/thirdweb/src/react/web/ui/components/buttons.tsx`, `packages/thirdweb/src/react/web/ui/ConnectWallet/screens/Buy/swap/ConfirmationScreen.tsx`, `packages/thirdweb/src/react/web/ui/ConnectWallet/screens/Buy/swap/PayWithCrypto.tsx`, `packages/thirdweb/src/react/web/ui/PayEmbed.test.tsx`, `packages/thirdweb/src/react/web/ui/ConnectWallet/screens/Buy/main/useBuyTxStates.ts`, `packages/thirdweb/src/react/web/ui/ConnectWallet/screens/Buy/PayTokenIcon.tsx`, `packages/thirdweb/src/react/web/ui/ConnectWallet/screens/Buy/main/useEnabledPaymentMethods.ts`, `packages/thirdweb/src/react/web/ui/ConnectWallet/screens/Buy/fiat/FiatSteps.tsx`, `packages/thirdweb/src/react/web/ui/ConnectWallet/screens/Buy/BuyScreen.tsx`, `pnpm-lock.yaml`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by MananTank** [comment]
* **#3421** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/3421?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @MananTank and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by MananTank** [comment]
I don't know why pnpm lock diff is so huge

**Comment by MananTank** [comment]
waiting for token icon to be added in quote/status endpoints so I can remove the temp fix from here

File: packages/thirdweb/src/react/web/ui/PayEmbed.test.tsx [hunk 1]
```diff
@@ -1,10 +1,34 @@
...
+  const { findByRole } = render(<PayEmbed client={TEST_CLIENT} />, {
+    setConnectedWallet: true,
+  });
+
+  // continue button is shown
+  const connectWalletButton = await findByRole("button", {
+    name: "Continue",
+  });
+
+  it("continue button is disabled if no token amount is set", async () => {
```
Comment (joaquim-verges) [inline packages/thirdweb/src/react/web/ui/PayEmbed.test.tsx:17]: nice!

**Review by joaquim-verges — APPROVED** [review]

**Pull request merged**